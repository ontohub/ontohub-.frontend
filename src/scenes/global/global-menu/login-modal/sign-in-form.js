import React from "react";
import { Formik } from "formik";
import { Button, Form, Message } from "semantic-ui-react";
import { ApolloConsumer, Mutation } from "react-apollo";
import { signIn, mutation } from "lib/session";

const initialValues = { username: "ada", password: "changemenow" };

export const SignInForm = ({ onSubmit, onSuccess, onError }) => (
  <ApolloConsumer>
    {client => (
      <Mutation mutation={mutation}>
        {mutate => (
          <Formik
            initialValues={initialValues}
            onSubmit={
              /* istanbul ignore next */ async (
                { username, password },
                { setErrors }
              ) => {
                try {
                  onSubmit();
                  const {
                    data: {
                      signIn: { jwt }
                    }
                  } = await mutate({
                    variables: { username, password }
                  });
                  signIn(client, jwt);
                  return onSuccess();
                } catch (e) {
                  setErrors({ password: "Incorrect username or password" });
                  return onError();
                }
              }
            }
          >
            {({ values, errors, handleChange, handleSubmit }) => (
              <Form error={!!errors.password} onSubmit={handleSubmit}>
                <Message error icon="warning sign" header={errors.password} />
                <Form.Group widths="equal">
                  <Form.Field error={!!errors.password}>
                    <label htmlFor="sign-in-username">Username</label>
                    <input
                      name="username"
                      id="sign-in-username"
                      type="text"
                      onChange={handleChange}
                      value={values.username}
                      placeholder="Username"
                    />
                  </Form.Field>
                  <Form.Field error={!!errors.password}>
                    <label htmlFor="sign-in-password">Password</label>
                    <input
                      name="password"
                      id="sign-in-password"
                      type="password"
                      onChange={handleChange}
                      value={values.password}
                      placeholder="Password"
                    />
                  </Form.Field>
                </Form.Group>
                <Button
                  type="submit"
                  id="sign-in-form-sign-in-button"
                  icon="sign in"
                  content="Sign in"
                  floated="right"
                  primary
                />
              </Form>
            )}
          </Formik>
        )}
      </Mutation>
    )}
  </ApolloConsumer>
);
