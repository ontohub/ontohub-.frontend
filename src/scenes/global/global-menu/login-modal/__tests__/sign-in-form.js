import React from "react";
import { SignInForm } from "../sign-in-form";

describe("SignInForm", () => {
  let wrapper;

  beforeAll(() => {
    wrapper = render(<SignInForm />);
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
