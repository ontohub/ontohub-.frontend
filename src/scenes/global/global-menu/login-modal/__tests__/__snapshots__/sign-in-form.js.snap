// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm matches the snapshot 1`] = `
<form
  class="ui form"
>
  <div
    class="ui error icon message"
  >
    <i
      aria-hidden="true"
      class="warning sign icon"
    >
      
    </i>
  </div>
  <div
    class="equal width fields"
  >
    <div
      class="field"
    >
      <label
        for="sign-in-username"
      >
        Username
      </label>
      <input
        id="sign-in-username"
        name="username"
        placeholder="Username"
        type="text"
        value="ada"
      >
        
      </input>
    </div>
    <div
      class="field"
    >
      <label
        for="sign-in-password"
      >
        Password
      </label>
      <input
        id="sign-in-password"
        name="password"
        placeholder="Password"
        type="password"
        value="changemenow"
      >
        
      </input>
    </div>
  </div>
  <button
    class="ui primary right floated button"
    id="sign-in-form-sign-in-button"
    role="button"
    type="submit"
  >
    <i
      aria-hidden="true"
      class="sign in icon"
    >
      
    </i>
    Sign in
  </button>
</form>
`;
